relay-common/src/utils.rs

```diff
@@ -40,9 +43,13 @@ impl Glob {
         pattern.push_str(&regex::escape(&glob[last..]));
         pattern.push('$');
 
+        // Prepare the new pattern replacing matches for `**` and `.` with non-capturing groups.
+        let replacer = pattern.replace("(.)", "(?:.)").replace("(.*?)", "(?:.*?)");
+
         Glob {
             value: glob.to_string(),
             pattern: Regex::new(&pattern).unwrap(),
+            replacer: Regex::new(&replacer).unwrap(),
```

Comment:
Thanks for pointing this!

It was a quick test on my side, to make sure the replacements and patterns are working. But you're definitely, it does not make any sense to keep 2 regexes inside the struct. I made the `Glob` configurable through the builder, with default behaviour like it was before (but it might be good idea to switch the builder around) and added customer deserialize function, see [b90a8ba](https://github.com/getsentry/relay/pull/1695/commits/b90a8bafd7f75cbc0f7ab8f3fbafd807ddd4025c)